hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -415,6 +447,34 @@ private void updateJMXParameters(
     }
   }
 
+  /**
+   * Get the HA status for a Namenode via RPC and add them to the report.
+   * @param report Namenode status report updating with HA status information data.
+   */
+  private void updateHAStatusParameters(NamenodeStatusReport report) {
+    if (localTarget != null) {
+      try {
+        // Determine if NN is active
+        if (localTargetHAProtocol == null) {
+          localTargetHAProtocol = localTarget.getHealthMonitorProxy(
+              conf, this.healthMonitorTimeoutMs);
+          LOG.debug("Get HA status with address {}", lifelineAddress);
+        }
+        HAServiceStatus status = localTargetHAProtocol.getServiceStatus();
+        report.setHAServiceState(status.getState());
+      } catch (Throwable e) {
+        if (e.getMessage().startsWith("HA for namenode is not enabled")) {
+          LOG.error("HA for {} is not enabled", getNamenodeDesc());
+          localTarget = null;
+        } else {
+          // Failed to fetch HA status, ignoring failure
+          LOG.error("Cannot fetch HA status for {}: {}", getNamenodeDesc(), e.getMessage(), e);
```

Comment:
And for logging exceptions, when do we need full stack? What doesn't need full stack? What do you think about this?

Reply:
Indeed, we can remove `e.getMessage()` in this place, because there is already a full stack. @goiri do you have any special considerations?

I personally feel that if we don't know the calling relationship of the exception very well and we need the full stack to located exceptions, or there is a very complex calling relationship, we can output the full stack. If we already know the calling process very well and the calling relationship is clear, we can just out put the message of exceptions, it will be clearer.